rdkit/Chem/fmcs/fmcs.py

```diff
@@ -743,20 +741,19 @@ def remove_unknown_bondtypes(typed_mol, supported_canonical_bondtypes):
 
 
 def find_upper_fragment_size_limits(rdmol, atoms):
-  max_num_atoms = max_twice_num_bonds = 0
+  max_num_atoms, max_twice_num_bonds = 0, 0
   for atom_indices in Chem.GetMolFrags(rdmol):
-    num_atoms = len(atom_indices)
-    if num_atoms > max_num_atoms:
-      max_num_atoms = num_atoms
+    max_num_atoms = max(max_num_atoms, len(atom_indices))
 
     # Every bond is connected to two atoms, so this is the
     # simplest way to count the number of bonds in the fragment.
     twice_num_bonds = 0
     for atom_index in atom_indices:
       # XXX Why is there no 'atom.GetNumBonds()'?
-      twice_num_bonds += sum(1 for bond in atoms[atom_index].GetBonds())
-    if twice_num_bonds > max_twice_num_bonds:
-      max_twice_num_bonds = twice_num_bonds
+      # Ichiru Take: len(atoms[atom_index].GetBonds()) would be more efficient but I don't know the input type.
+      twice_num_bonds += sum(1 for bond in atoms[atom_index].GetBonds()) 
```

Comment:
@greglandrum what do you think?

Reply:
I have tested `len(atoms[atom_index].GetBonds())` and it worked